x-pack/plugin/esql/compute/src/main/generated-src/org/elasticsearch/compute/aggregation/StdDeviationFloatAggregator.java

```diff
@@ -0,0 +1,234 @@
+                final var st = groupId < state.states.size() ? state.states.get(groupId) : null;
+                if (st != null) {
+                    meanBuilder.appendDouble(st.mean());
+                    m2Builder.appendDouble(st.m2());
+                    countBuilder.appendLong(st.count());
+                } else {
+                    meanBuilder.appendNull();
+                    m2Builder.appendNull();
+                    countBuilder.appendNull();
+                }
+            }
+            blocks[offset + 0] = meanBuilder.build();
+            blocks[offset + 1] = m2Builder.build();
+            blocks[offset + 2] = countBuilder.build();
+        }
+    }
+
+    public static Block evaluateFinal(GroupingStdDeviationFloatState state, IntVector selected, DriverContext driverContext) {
+        try (DoubleBlock.Builder builder = driverContext.blockFactory().newDoubleBlockBuilder(selected.getPositionCount())) {
+            for (int i = 0; i < selected.getPositionCount(); i++) {
+                final var groupId = selected.getInt(i);
+                final var st = groupId < state.states.size() ? state.states.get(groupId) : null;
+                if (st != null) {
+                    final var m2 = st.m2();
+                    if (Double.isFinite(m2) == false) {
+                        builder.appendNull();
+                    } else {
+                        builder.appendDouble(st.evaluateFinal());
+                    }
+                } else {
+                    builder.appendNull();
+                }
+            }
+            return builder.build();
+        }
+    }
+
+    static final class StdDeviationFloatState implements AggregatorState {
+
+        private WelfordAlgorithm welfordAlgorithm;
+
+        StdDeviationFloatState() {
+            this(0, 0, 0);
+        }
+
+        StdDeviationFloatState(double mean, double m2, long count) {
+            this.welfordAlgorithm = new WelfordAlgorithm(mean, m2, count);
+        }
+
+        public void add(float value) {
+            welfordAlgorithm.add(value);
+        }
+
+        public void combine(double mean, double m2, long count) {
+            welfordAlgorithm.add(mean, m2, count);
+        }
+
+        @Override
+        public void toIntermediate(Block[] blocks, int offset, DriverContext driverContext) {
+            StdDeviationFloatAggregator.evaluateIntermediate(this, driverContext, blocks, offset);
+        }
+
+        @Override
+        public void close() {}
+
+        public double mean() {
+            return welfordAlgorithm.mean();
+        }
+
+        public double m2() {
+            return welfordAlgorithm.m2();
+        }
+
+        public long count() {
+            return welfordAlgorithm.count();
+        }
+
+        public double evaluateFinal() {
+            return welfordAlgorithm.evaluate();
+        }
+    }
+
+    static final class GroupingStdDeviationFloatState implements GroupingAggregatorState {
```

Comment:
It seems the state and grouping state are the same for all data types. Could we use one class and share it between them instead?